### PR TITLE
[QOL-7905] patch CKAN 2.9 so password resets work with restricted user API

### PR DESCRIPTION
**Pull request by ThrawnCA**

### CODE REVIEW

**Review by duttonw — APPROVED** [review]

File: test/features/organisations.feature [hunk 1]
```diff
@@ -0,0 +1,53 @@
+@users
+Feature: Organization APIs
+
+    Scenario: Organisation membership is accessible to sysadmins
```
Comment (duttonw) [inline test/features/organisations.feature:4]: these 3 could be done with 'examples' to swap the persona. see https://github.com/salsadigitalauorg/ckanext-data-qld/blob/40539ba63a2cfc523039bb7434ba07111ef1b356/test/features/resource_availability.feature#L95

File: test/features/resources.feature [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+        Examples:
+        | name | url |
+        | Good link | http://www.qld.gov.au |
+        | Good IP address | http://1.2.3.4 |
+        | Domain starting with numbers | http://1.2.3.4.example.com |
+        | Domain ending with numbers | http://example.com.1.2.3.4 |
+        | Domain ending with private | http://example.com.private |
+
+    Scenario: Link resource with missing or invalid protocol should use HTTP
```
Comment (duttonw) [inline test/features/resources.feature:19]: could be a Examples test like above with expected output and more than 1 test.

File: test/features/users.feature [hunk 3]
```diff
@@ -0,0 +1,173 @@
+@users
+Feature: User APIs
+
+    Scenario: User autocomplete is accessible to sysadmins
```
Comment (duttonw) [inline test/features/users.feature:4]: Scenario Outline with Example 'personas' could collapse these similar tests.

**Review by duttonw — APPROVED** [review]
looking good, couple of tidy up comments which can be done if time permits.

**Review by duttonw — APPROVED** [review]

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -32,7 +32,7 @@ jobs:
 
       - name: Test
         run: .circleci/test.sh
-        timeout-minutes: 25
+        timeout-minutes: 30
```
Comment (duttonw) [inline .github/workflows/test.yml:35]: Might be time to look into running tests in parallel to claw back our turnaround time

Comment (ThrawnCA) [reply .github/workflows/test.yml:35]: GitHub Actions doesn't seem to have a native option. Might be able to do something with https://github.com/hugeinc/behave-parallel ?

**Pull request merged**